packages/tools/kadena-cli/src/services/pactjs/pactjs.service.ts

```diff
@@ -0,0 +1,227 @@
+import type { ChainId } from '@kadena/client';
+import { Pact } from '@kadena/client';
+import { generateDts, pactParser } from '@kadena/pactjs-generator';
+import { EOL } from 'os';
+import { dirname, join } from 'path';
+import * as prettier from 'prettier';
+import { rimraf } from 'rimraf';
+import { log } from '../../utils/logger.js';
+import type { Services } from '../index.js';
+import type { IContractGenerateOptions } from './pactjs.types.js';
+import { fetchModule } from './utils/callLocal.js';
+import { extractImportedFiles } from './utils/extractImportedFiles.js';
+import { networkMap } from './utils/networkMap.js';
+import { shallowFindFile } from './utils/shallowFindFile.js';
+import { verifyTsconfigTypings } from './utils/verifyTsconfigTypings.js';
+
+export interface IPactJSService {
+  retrieveContractFromChain: (
+    module: string,
+    apiHost: string,
+    chain: number | string,
+    network: string,
+  ) => Promise<string>;
+  generate: (args: IContractGenerateOptions) => Promise<void>;
+}
+
+export class PactJSService implements IPactJSService {
+  // eslint-disable-next-line @typescript-eslint/naming-convention, @typescript-eslint/parameter-properties
+  public constructor(private services: Services) {}
+
+  public async retrieveContractFromChain(
+    module: string,
+    apiHost: string,
+    chain: number | string,
+    network: string,
+  ): Promise<string> {
+    const command = Pact.builder
+      .execution(`(describe-module "${module}")`)
+      .setNetworkId(
+        network === 'testnet' || network === 'mainnet'
+          ? networkMap[network].network
+          : network,
+      )
+      .setMeta({ chainId: chain.toString() as ChainId })
+      .createTransaction();
+
+    const { code, error } = await fetchModule(apiHost, JSON.stringify(command));
+
+    if (error !== undefined) {
+      log.error(
+        `Could not retrieve ${module} from network:${network} - chain:${chain}\nerror: ${error}`,
+      );
+      throw new Error(
+        `Could not retrieve ${module} from network:${network} - chain:${chain}\nerror: ${error}`,
+      );
+    }
+
+    return code;
+  }
+
+  public async generate(args: IContractGenerateOptions): Promise<void> {
+    const targetPackageJson = await shallowFindFile(
+      process.cwd(),
+      'package.json',
+    );
+
+    if (targetPackageJson === null || targetPackageJson === undefined) {
+      throw new Error('Could not find package.json');
+    }
+
+    const moduleDtss = await this._generator(args);
+
+    const targetDirectory = join(
+      dirname(targetPackageJson),
+      'node_modules',
+      '.kadena/pactjs-generated',
+    );
+
+    if (args.clean === true) {
+      log.info(`Cleaning ${targetDirectory}`);
+      rimraf.sync(targetDirectory);
```

Comment:
should be replaced with `filesystem` service. that will also save us using another dependency :) It's just a `deleteDirectory` which it already has.